backend/client.go

```diff
@@ -194,9 +207,32 @@ func (c *backendClient) ResumeOrchestration(ctx context.Context, id api.Instance
 //
 // [api.ErrInstanceNotFound] is returned if the specified orchestration instance doesn't exist.
 // [api.ErrNotCompleted] is returned if the specified orchestration instance is still running.
-func (c *backendClient) PurgeOrchestrationState(ctx context.Context, id api.InstanceID) error {
+func (c *backendClient) PurgeOrchestrationState(ctx context.Context, id api.InstanceID, opts ...api.PurgeOptions) error {
+	req := &protos.PurgeInstancesRequest{Request: &protos.PurgeInstancesRequest_InstanceId{InstanceId: string(id)}, Recursive: true}
+	for _, configure := range opts {
+		if err := configure(req); err != nil {
+			return fmt.Errorf("failed to configure purge request: %w", err)
+		}
+	}
+	subOrchestrationInstances := make([]api.InstanceID, 0, 10)
+	var err error
+	if req.Recursive {
+		// Fetching sub-orchestration instances
+		subOrchestrationInstances, err = GetSubOrchestrationInstances(ctx, c.be, id, true)
+		if err != nil {
+			return fmt.Errorf("failed to fetch sub-orchestration instances: %w", err)
```

Comment:
Looking at the implementation of `GetSubOrchestrationInstances`, it seems you'll return an error if a sub-orchestration is in a running state or if some orchestration doesn't exist. The error message then becomes misleading because we didn't actually fail to fetch sub-orchestrations (we failed the operation for some other reason). I think you should either return `err` as-is or return an error indicating that the *purge* operation failed. and have `err` include the inner details.

Reply:
Modified